src/Core/Models/Api/Request/Accounts/RegisterRequestModel.cs

```diff
@@ -25,7 +27,7 @@ public class RegisterRequestModel : IValidatableObject
         public Guid? OrganizationUserId { get; set; }
         public KdfType? Kdf { get; set; }
         public int? KdfIterations { get; set; }
-        public string ReferenceId { get; set; }
+        public ReferenceEventData ReferenceData { get; set; }
```

Comment:
I don't really like coupling data and API models. Even though they are the same in this case. The API interface should always have independent request and response models.

Reply:
You can just add the API request model for this as a subclass to this model. Or treat reference data more generically as a <string, object> dictionary or something.